demo/java-demo-client/src/main/java/org/cloudfoundry/abacus/demo/DemoApplication.java

```diff
@@ -0,0 +1,87 @@
+package org.cloudfoundry.abacus.demo;
+
+import org.cloudfoundry.abacus.demo.model.*;
+import org.cloudfoundry.abacus.demo.model.AggregatedUsage.WindowName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URI;
+import java.util.List;
+
+@SpringBootApplication
+public class DemoApplication implements ApplicationRunner {
+
+  private static final Logger log = LoggerFactory.getLogger(DemoApplication.class);
+
+  @Value("${collector:http://localhost:9080}")
+  private String collector;
+
+  @Value("${reporting:http://localhost:9088}")
+  private String reporting;
+
+  public static void main(String... args) throws Exception {
+    SpringApplication.run(DemoApplication.class, args);
+  }
+
+  @Override
+  public void run(ApplicationArguments args) throws Exception {
+    RestTemplate restTemplate = new RestTemplate();
+
+    postUsage(restTemplate);
+
+    Thread.sleep(2000);
+
+    Report report = getReport(restTemplate);
+
+    int calls = getMonthlyHeavyAPICalls(report);
+    log.info("Monthly heavy API calls {}", calls);
+
+    System.exit(0);
```

Comment:
Is `System.exit` necessary? Doesn't the app exit on its own?

Reply:
As this is Spring boot app it is kept up&running. You need to call the shutdown actuator endpoint (disabled by default). Since this is meant to be example/demo console app I decided to add System.exit to simplify testing and avoid questions like "./gradlew bootRun hangs at 75%"